tts/tts/src/durability.rs

```diff
@@ -0,0 +1,1806 @@
+            }
+        }
+
+        fn validate_input(
+            input: TextInput,
+            voice: crate::exports::golem::tts::voices::VoiceBorrow<'_>,
+        ) -> Result<ValidationResult, TtsError> {
+            init_logging();
+
+            let durability = Durability::<ValidationResultOutput, TtsError>::new(
+                "golem_tts",
+                "validate_input",
+                DurableFunctionType::WriteRemote,
+            );
+            if durability.is_live() {
+                let result = Impl::validate_input(input.clone(), voice);
+                let result = result.map(|v| ValidationResultOutput { result: v });
+                durability
+                    .persist(ValidateInputInput { input }, result)
+                    .map(|output| output.result)
+            } else {
+                durability
+                    .replay()
+                    .map(|output: ValidationResultOutput| output.result)
+            }
+        }
+    }
+
+    impl<Impl: ExtendedGuest> StreamingGuest for DurableTts<Impl> {
+        type SynthesisStream = Impl::SynthesisStream;
+        type VoiceConversionStream = Impl::VoiceConversionStream;
+
+        fn create_stream(
+            voice: crate::exports::golem::tts::voices::VoiceBorrow<'_>,
+            options: Option<SynthesisOptions>,
+        ) -> Result<SynthesisStream, TtsError> {
+            init_logging();
+            Impl::create_stream(voice, options)
+        }
+
+        fn create_voice_conversion_stream(
+            target_voice: crate::exports::golem::tts::voices::VoiceBorrow<'_>,
+            options: Option<SynthesisOptions>,
+        ) -> Result<VoiceConversionStream, TtsError> {
+            init_logging();
+            Impl::create_voice_conversion_stream(target_voice, options)
+        }
+    }
+
+    #[allow(dead_code)]
+    enum DurableVoiceConversionStreamState<Impl: ExtendedGuest> {
+        Live {
+            stream: Impl::VoiceConversionStream,
+            pollables: Vec<LazyInitializedPollable>,
+        },
+        Replay {
+            options: Box<Option<SynthesisOptions>>,
+            pollables: Vec<LazyInitializedPollable>,
+            partial_result: Vec<AudioChunk>,
+            #[allow(dead_code)]
+            finished: bool,
+        },
+    }
+
+    #[allow(dead_code)]
+    pub struct DurableVoiceConversionStream<Impl: ExtendedGuest> {
+        state: RefCell<Option<DurableVoiceConversionStreamState<Impl>>>,
+        subscription: RefCell<Option<Pollable>>,
+    }
+
+    impl<Impl: ExtendedGuest> Drop for DurableVoiceConversionStream<Impl> {
+        fn drop(&mut self) {
+            let _ = self.subscription.take();
+            match self.state.take() {
+                Some(DurableVoiceConversionStreamState::Live {
+                    mut pollables,
+                    stream,
+                }) => {
+                    with_persistence_level(PersistenceLevel::PersistNothing, move || {
```

Comment:
This `with_persistence_level` wrapper in a `drop` is weird, what was the intention?